grapple/types/redirects.py

```diff
@@ -1,31 +1,52 @@
+import json
+
+from typing import Optional
+
 import graphene
 
-from django.conf import settings
 from wagtail.contrib.redirects.models import Redirect
+from wagtail.models import Page, Site
 
 from .pages import get_page_interface
 
 
 class RedirectType(graphene.ObjectType):
     old_path = graphene.String(required=True)
     old_url = graphene.String(required=True)
-    new_url = graphene.String(required=True)
+    new_url = graphene.String(required=False)
     page = graphene.Field(get_page_interface())
     is_permanent = graphene.Boolean(required=True)
 
-    # Give old_path with BASE_URL attached.
-    def resolve_old_url(self, info, **kwargs):
-        return settings.BASE_URL + self.old_path
+    def resolve_old_url(self, info, **kwargs) -> str:
+        """
+        If the redirect is for a specific site, append hostname and port to path.
+        Otherwise, return a JSON string representing a list of all site urls.
+        """
+        if self.site:
+            return f"http://{self.site.hostname}:{self.site.port}/{self.old_path}"
```

Comment:
**Issue:** Not all sites will start with "http://", so this value is erroneous. Wagtail editors can specify "https://" for a `Site` by changing the port to 443.

Reply:
Thanks! Addressed in 33561505891fe6ec7d6a5901f9301af1d26aaad5